chia/cmds/wallet_funcs.py

```diff
@@ -170,6 +173,167 @@ async def add_token(args: dict, wallet_client: WalletRpcClient, fingerprint: int
+                    print(f"Created offer with ID {trade_record.trade_id}")
+                    print(f"Use chia wallet get_offers --id {trade_record.trade_id} -f {fingerprint} to view status")
+                else:
+                    print("Error creating offer")
+
+
+def timestamp_to_time(timestamp):
+    return datetime.fromtimestamp(timestamp).strftime("%Y-%m-%d %H:%M:%S")
+
+
+async def print_offer_summary(wallet_client: WalletRpcClient, sum_dict: dict):
+    for asset_id, amount in sum_dict.items():
+        if asset_id == "xch":
+            wid: str = "1"
+            name: str = "XCH"
+            unit: int = units["chia"]
+        else:
+            result = await wallet_client.cat_asset_id_to_name(bytes.fromhex(asset_id))
+            wid = "Unknown"
+            name = asset_id
+            unit = units["colouredcoin"]
+            if result is not None:
+                wid = str(result[0])
+                name = result[1]
+        print(f"    - {name} (Wallet ID: {wid}): {Decimal(int(amount)) / unit}")
+
+
+async def print_trade_record(record, wallet_client: WalletRpcClient, summaries: bool = False) -> None:
+    print()
+    print(f"Record with id: {record.trade_id}")
+    print("---------------")
+    print(f"Created at: {timestamp_to_time(record.created_at_time)}")
+    print(f"Confirmed at: {record.confirmed_at_index}")
+    print(f"Accepted at: {timestamp_to_time(record.accepted_at_time) if record.accepted_at_time else 'N/A'}")
+    print(f"Status: {TradeStatus(record.status).name}")
+    if summaries:
+        print("Summary:")
+        offer = Offer.from_bytes(record.offer)
+        offered, requested = offer.summary()
+        print("  OFFERED:")
+        await print_offer_summary(wallet_client, offered)
+        print("  REQUESTED:")
+        await print_offer_summary(wallet_client, requested)
+        print("Pending Balances:")
+        await print_offer_summary(wallet_client, offer.get_pending_amounts())
+        print(f"Fees: {Decimal(offer.bundle.fees()) / units['chia']}")
+    print("---------------")
+
+
+async def get_offers(args: dict, wallet_client: WalletRpcClient, fingerprint: int) -> None:
+    id, filepath, all, summaries = tuple(list(args.values())[0:4])
+    file_contents: bool = (filepath is not None) or summaries
+    if id is None:
+        records = await wallet_client.get_all_offers(file_contents)
+    else:
+        records = [await wallet_client.get_offer(hexstr_to_bytes(id), file_contents)]
+        if filepath is not None:
+            with open(pathlib.Path(filepath), "w") as file:
+                file.write(records[0].offer.hex())
+                file.close()
+
+    for record in records:
+        if "PENDING" in TradeStatus(record.status).name or all or id is not None:
+            await print_trade_record(record, wallet_client, summaries=summaries)
+
+
+async def take_offer(args: dict, wallet_client: WalletRpcClient, fingerprint: int) -> None:
+    if "." in args["file"]:
```

Comment:
is this to check for a filename with an extension (.offer)? I might add a comment to that effect. Also, if a filename is provided without a `.`, the error message is pretty cryptic:

`Exception from 'wallet' non-hexadecimal number found in fromhex() arg at position 0`

Can we inform the user that the offer file should contain the `.offer` extension? Or emit a more informative error.